Rust stable support for 1.27+

Rust 1.27 will be released momentarily. It includes a stable API for SIMD on x86/x64:

https://github.com/rust-lang/rust/blob/f8c8701b18b88ca3a96b2cf83f76a12fb1426e74/RELEASES.md#libraries

miscreant.rs (master) should be nearly ready to support stable. Here's a quick checklist:

* [x] Add `stable` to the list of Rust CI targets
* [x] Ensure Miscreant builds on Rust 1.27
* [ ] Ensure documentation consistently reflects `stable` Rust support (and the needed `RUSTFLAGS`)

Currently it seems to fail to compile one of the dependencies because of a `feature` flag that's used:

```
error[E0554]: #![feature] may not be used on the stable release channel
 --> /home/spacekookie/.cargo/registry/src/github.com-1ecc6299db9ec823/clear_on_drop-0.2.3/src/lib.rs:2:34
  |
2 | #![cfg_attr(feature = "nightly", feature(asm, i128_type, specialization))]
  |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0554`.
error: Could not compile `clear_on_drop`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/rust/README.md
+++ b/rust/README.md
@@ -48,17 +48,15 @@ For more information, see the [toplevel README.md].
 
 ## Requirements
 
-miscreant.rs presently requires the following:
+miscreant.rs works on stable rust since `1.27`. By default it is built with aesni
+support which requires an x86 instruction set. You can disable this with the `aes-soft`
+feature flag which enables usage on other architectures.
 
-* **x86_64** CPU architecture
-* Rust **nightly** compiler
+The default configuration uses the `core::arch` API for stable access to 
+CPU intrinsics, namely the [Intel AES-NI]  instructions which provide a 
+hardware implementation of AES.
 
-This is because it depends on the `aesni` crate which uses the `core::arch` API
-for (soon-to-be) stable access to CPU intrinsics, namely the [Intel AES-NI] CPU
-instructions which provide a hardware implementation of AES.
-
-To access these features, you will need both a relatively recent
-Rust nightly and to pass the following as RUSTFLAGS:
+To access these features, you will need to pass the following as RUSTFLAGS:
 
 ```
 RUSTFLAGS=-Ctarget-feature=+aes
@@ -71,7 +69,6 @@ You can configure your `~/.cargo/config` to always pass these flags:
 rustflags = ["-Ctarget-feature=+aes"]
 ```
 
-[aesni]: https://github.com/RustCrypto/block-ciphers
 [Intel AES-NI]: https://software.intel.com/en-us/blogs/2012/01/11/aes-ni-in-laymens-terms
 
 ## Help and Discussion

--- a/rust/src/lib.rs
+++ b/rust/src/lib.rs
@@ -3,8 +3,14 @@
 //!
 //! # Build Notes
 //!
-//! This crate depends on the `aesni` crate, which uses the new `core::arch`
-//! API to invoke CPU instructions for performing AES in hardware.
+//! miscreant.rs works on stable rust since `1.27`. By default it is built with aesni
+//! support which requires an x86 instruction set. You can disable this with 
+//! the `aes-soft` feature flag which enables usage on other architectures.
+//! 
+//! The default configuration uses the `core::arch` API for stable access to 
+//! CPU intrinsics, namely the [Intel AES-NI]  instructions which provide a 
+//! hardware implementation of AES.
+//! 
 //!
 //! To access these features, you will need both a relatively recent
 //! Rust nightly and to pass the following as RUSTFLAGS:

